### ISSUE TITLE
Auto-install recommended extensions

### ISSUE THREAD

**Issue by arcanis** [microsoft/vscode-remote-release#6364]
I noticed my extensions defined in `extensions.json` were not installed when rebuilding the codespace. Instead, I have to duplicate them into `devcontainer.json`. It'd be nice if this list could be shared (perhaps by allowing `"recommended"` as a valid extension name in the devcontainer file, which would expand to all recommended extensions?).

**Comment by pamelafox** [comment]
We are experiencing an issue where Codespaces gives the recommended pop-up for an extension that was already installed in devcontainer.json but is also listed in extensions.json-
https://github.com/Azure/azure-dev/issues/951
Is the best practice to only be listing it in devcontainer.json? Or is it a bug in VS Code that it pops up the recommendeded dialog?

![Screen Shot 2022-10-24 at 3 26 54 PM](https://user-images.githubusercontent.com/297042/197641716-3871a43f-d14c-4965-ad1d-0355409f5cdb.png)

**Comment by sanmai-NL** [comment]
This would help, but https://code.visualstudio.com/docs/editor/extension-marketplace#_command-line-extension-management does not work inside an attached container:

```console
# code --list-extensions '@recommended'
Unable to connect to VS Code server: Error in request.
Error: connect ENOENT /tmp/vscode-ipc-8742bf1b-bfdb-4227-8b77-a0852d28a62e.sock
    at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1157:16) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'connect',
  address: '/tmp/vscode-ipc-8742bf1b-bfdb-4227-8b77-a0852d28a62e.sock'
}
# code --show-versions
Unable to connect to VS Code server: Error in request.
Error: connect ENOENT /tmp/vscode-ipc-8742bf1b-bfdb-4227-8b77-a0852d28a62e.sock
    at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1157:16) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'connect',
  address: '/tmp/vscode-ipc-8742bf1b-bfdb-4227-8b77-a0852d28a62e.sock'
}
```

**Comment by Destroy666x** [comment]
Yes please. Also, after it's implemented, there needs to be some separation of recommended and required/autoinstalled extensions. E.g. let's say 80% of my recommendations should be autoinstalled, but 20% could be skipped because they're much more of a personal preference that I'd like to keep using inside dev containers.

**Comment by Malix-off** [comment]
Just had this thought too

It would be nice to have it, since it is not DRY and makes us dependant to support **two** standard behavior for sharing extensions:
- Using Dev Containers: `.devcontainer/devcontainer.json`: `customizations.vscode.extensions`
- Not Using Dev Containers: `.vscode/extensions.json`: `recommendations`

**Comment by Malix-off** [comment]
VSCode doesn't support "required" extensions, only "recommended".

As such, I think that to make a separation between both for devcontainer, a proposal should be accepted to make VScode support "required" extensions.
